htdocs/admin/invoice_situation.php

```diff
@@ -79,12 +79,13 @@
 $item->nameText = $langs->trans('AllowedInvoiceForRetainedWarranty');
 
 $arrayAvailableType = array(
+	0 => "", // don't use $showempty in selectArray since it sets a value of -1
```

Comment:
We must use showempty into select component when we want empty value. Empty value generates troubles depending on js code used to beautify the combi list (like select2).
So fix must be managed during or before  the insert. For example if value is -1, no update should be done.

Reply:
I changed the PR to solve a related issue on display of chosen value (see initial message edit).